backbone.js

```diff
@@ -159,11 +109,9 @@
     trigger: function(name) {
       if (!this._events) return this;
       var args = slice.call(arguments, 1);
-      if (!eventsApi(this, 'trigger', name, args)) return this;
-      var events = this._events[name];
-      var allEvents = this._events.all;
-      if (events) triggerEvents(events, args);
-      if (allEvents) triggerEvents(allEvents, arguments);
+
+      // Use `eventsApi` to normalize `name` into the proper event names.
```

Comment:
I agree the source could be better arranged, I'll work on that tonight along with the comments.

> There's so much extra overhead and indirection going on here

I actually thought it was kind of easier to follow without the old `eventsApi` behavior, but then I had to account for `off` cleaning up any listeners. It makes it easier in my mind to think of all the `*Api` functions as special purpose reduce iteratees, with `eventsApi` being `_.reduce`.